File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
      
        if not head:
            return None
        
        lastNode = head
        length = 1
        while lastNode.next:
          lastNode = lastNode.next
          length += 1
        
        lastNode.next = head
        k = k % length
        
        temp = head
        for _ in range(length - k - 1):
          temp = temp.next
          
        rotated_node = temp.next
        temp.next = None
        
        return rotated_node
        
        

File: 0061-rotate-list/0061-rotate-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */

//   1  ->   2  ->   3  ->   4  ->   5  ->  1  ->  2  ->  3  ->  4  ->  5
//  curr     

var rotateRight = function(head, k) {
  if (!head) return null;
  
  let lastNode = head;
  let length = 1;
  while (lastNode.next) {
    lastNode = lastNode.next;
    length++;
  }
  
  lastNode.next = head;
  k = k % length;

  let temp = head;
  for (let i = 1; i < length - k; i++) {
    temp = temp.next;
  }
  
  let rotatedNode = temp.next;
  temp.next = null;
  return rotatedNode;  
};
